### ISSUE TITLE
All entities are marked as "no longer provided by the integration"

### ISSUE THREAD

**Issue by ThomasBergholdWieser** [signalkraft/mypyllant-component#126]
### Before submitting a new issue

- [x] I [redownloaded the latest release](https://my.home-assistant.io/redirect/hacs_repository/?owner=signalkraft&repository=mypyllant-component) and restarted Home Assistant
- [x] I've checked [known issues](https://signalkraft.com/mypyllant-component/#known-issues)
- [x] I've [enabled debug logs](https://signalkraft.com/mypyllant-component/3-contributing/#debugging) and attached the relevant logs to this ticket

### Problem description

All entities of the integration are marked as no longer provided by it, therefore making it unusable. The official app itself still works, so the system seems to be online and working fine. It seems to be related to the integration.

### Logs

```
2024-03-05 08:34:23.658 WARNING (MainThread) [homeassistant.bootstrap] Waiting on integrations to complete setup: mypyllant
2024-03-05 08:34:37.532 ERROR (MainThread) [custom_components.mypyllant.coordinator] Error fetching myVAILLANT data: 500, message='Internal Server Error', url=URL('https://api.vaillant-group.com/service-connected-control/end-user-app-api/v1/emf/v2/899af599-23e9-46b6-9988-3cc3e37cb006/currentSystem')
2024-03-05 08:35:07.791 ERROR (MainThread) [custom_components.mypyllant.coordinator] Error fetching myVAILLANT data: 500, message='Internal Server Error', url=URL('https://api.vaillant-group.com/service-connected-control/end-user-app-api/v1/emf/v2/899af599-23e9-46b6-9988-3cc3e37cb006/currentSystem')
2024-03-05 08:35:07.872 WARNING (MainThread) [custom_components.mypyllant.binary_sensor] No system data, skipping binary sensors
2024-03-05 08:35:07.873 WARNING (MainThread) [custom_components.mypyllant.calendar] No system data, skipping calendar entities
2024-03-05 08:35:07.878 WARNING (MainThread) [custom_components.mypyllant.number] No system data, skipping number entities
2024-03-05 08:35:07.878 WARNING (MainThread) [custom_components.mypyllant.sensor] No system data, skipping sensors
2024-03-05 08:35:07.879 WARNING (MainThread) [custom_components.mypyllant.sensor] No daily data, skipping sensors
2024-03-05 08:35:07.879 WARNING (MainThread) [custom_components.mypyllant.switch] No system data, skipping switch entities
2024-03-05 08:35:07.955 WARNING (MainThread) [custom_components.mypyllant.climate] No system data, skipping climate
2024-03-05 08:35:07.957 WARNING (MainThread) [custom_components.mypyllant.datetime] No system data, skipping date time entities
2024-03-05 08:35:07.960 WARNING (MainThread) [custom_components.mypyllant.water_heater] No system data, skipping water heater
2024-03-05 08:36:14.894 ERROR (MainThread) [custom_components.mypyllant.coordinator] Error fetching myVAILLANT data: 500, message='Internal Server Error', url=URL('https://api.vaillant-group.com/service-connected-control/end-user-app-api/v1/emf/v2/899af599-23e9-46b6-9988-3cc3e37cb006/currentSystem')
2024-03-05 08:36:45.163 ERROR (MainThread) [custom_components.mypyllant.coordinator] Error fetching myVAILLANT data: 500, message='Internal Server Error', url=URL('https://api.vaillant-group.com/service-connected-control/end-user-app-api/v1/emf/v2/899af599-23e9-46b6-9988-3cc3e37cb006/currentSystem')
2024-03-05 08:36:45.169 WARNING (MainThread) [custom_components.mypyllant.binary_sensor] No system data, skipping binary sensors
2024-03-05 08:36:45.169 WARNING (MainThread) [custom_components.mypyllant.calendar] No system data, skipping calendar entities
2024-03-05 08:36:45.170 WARNING (MainThread) [custom_components.mypyllant.climate] No system data, skipping climate
2024-03-05 08:36:45.170 WARNING (MainThread) [custom_components.mypyllant.datetime] No system data, skipping date time entities
2024-03-05 08:36:45.170 WARNING (MainThread) [custom_components.mypyllant.number] No system data, skipping number entities
2024-03-05 08:36:45.171 WARNING (MainThread) [custom_components.mypyllant.sensor] No system data, skipping sensors
2024-03-05 08:36:45.171 WARNING (MainThread) [custom_components.mypyllant.sensor] No daily data, skipping sensors
2024-03-05 08:36:45.171 WARNING (MainThread) [custom_components.mypyllant.switch] No system data, skipping switch entities
2024-03-05 08:36:45.172 WARNING (MainThread) [custom_components.mypyllant.water_heater] No system data, skipping water heater
2024-03-05 08:43:48.912 ERROR (MainThread) [custom_components.mypyllant.coordinator] Error fetching myVAILLANT data: 500, message='Internal Server Error', url=URL('https://api.vaillant-group.com/service-connected-control/end-user-app-api/v1/emf/v2/899af599-23e9-46b6-9988-3cc3e37cb006/currentSystem')
2024-03-05 08:44:19.170 ERROR (MainThread) [custom_components.mypyllant.coordinator] Error fetching myVAILLANT data: 500, message='Internal Server Error', url=URL('https://api.vaillant-group.com/service-connected-control/end-user-app-api/v1/emf/v2/899af599-23e9-46b6-9988-3cc3e37cb006/currentSystem')
2024-03-05 08:44:19.175 WARNING (MainThread) [custom_components.mypyllant.binary_sensor] No system data, skipping binary sensors
2024-03-05 08:44:19.175 WARNING (MainThread) [custom_components.mypyllant.calendar] No system data, skipping calendar entities
2024-03-05 08:44:19.176 WARNING (MainThread) [custom_components.mypyllant.climate] No system data, skipping climate
2024-03-05 08:44:19.176 WARNING (MainThread) [custom_components.mypyllant.datetime] No system data, skipping date time entities
2024-03-05 08:44:19.176 WARNING (MainThread) [custom_components.mypyllant.number] No system data, skipping number entities
2024-03-05 08:44:19.177 WARNING (MainThread) [custom_components.mypyllant.sensor] No system data, skipping sensors
2024-03-05 08:44:19.177 WARNING (MainThread) [custom_components.mypyllant.sensor] No daily data, skipping sensors
2024-03-05 08:44:19.177 WARNING (MainThread) [custom_components.mypyllant.switch] No system data, skipping switch entities
2024-03-05 08:44:19.177 WARNING (MainThread) [custom_components.mypyllant.water_heater] No system data, skipping water heater
2024-03-05 08:49:56.782 DEBUG (MainThread) [custom_components.mypyllant] Starting mypyllant component v0.7.3 (library 0.7.26) with homeassistant 2024.2.5, pydantic 1.10.12, and aiohttp 3.9.3
2024-03-05 08:49:56.783 DEBUG (MainThread) [custom_components.mypyllant] Creating API and logging in with <email removed> in realm <realm removed>
2024-03-05 08:49:58.065 DEBUG (MainThread) [custom_components.mypyllant] Refreshing SystemCoordinator
2024-03-05 08:49:58.065 DEBUG (MainThread) [custom_components.mypyllant.coordinator] Starting async update data for SystemCoordinator
2024-03-05 08:49:58.065 DEBUG (MainThread) [custom_components.mypyllant.coordinator] Waiting 118s until token refresh for <email removed>
2024-03-05 08:50:28.447 ERROR (MainThread) [custom_components.mypyllant.coordinator] Error fetching myVAILLANT data: 500, message='Internal Server Error', url=URL('https://api.vaillant-group.com/service-connected-control/end-user-app-api/v1/emf/v2/899af599-23e9-46b6-9988-3cc3e37cb006/currentSystem')
2024-03-05 08:50:28.447 DEBUG (MainThread) [custom_components.mypyllant.coordinator] Finished fetching myVAILLANT data in 30.383 seconds (success: False)
2024-03-05 08:50:28.448 DEBUG (MainThread) [custom_components.mypyllant] Refreshing DailyDataCoordinator
2024-03-05 08:50:28.448 DEBUG (MainThread) [custom_components.mypyllant.coordinator] Starting async update data for DailyDataCoordinator
2024-03-05 08:50:28.449 DEBUG (MainThread) [custom_components.mypyllant.coordinator] Waiting 88s until token refresh for <email removed>
2024-03-05 08:50:58.702 ERROR (MainThread) [custom_components.mypyllant.coordinator] Error fetching myVAILLANT data: 500, message='Internal Server Error', url=URL('https://api.vaillant-group.com/service-connected-control/end-user-app-api/v1/emf/v2/899af599-23e9-46b6-9988-3cc3e37cb006/currentSystem')
2024-03-05 08:50:58.703 DEBUG (MainThread) [custom_components.mypyllant.coordinator] Finished fetching myVAILLANT data in 30.255 seconds (success: False)
2024-03-05 08:50:58.712 WARNING (MainThread) [custom_components.mypyllant.binary_sensor] No system data, skipping binary sensors
2024-03-05 08:50:58.712 WARNING (MainThread) [custom_components.mypyllant.calendar] No system data, skipping calendar entities
2024-03-05 08:50:58.713 WARNING (MainThread) [custom_components.mypyllant.climate] No system data, skipping climate
2024-03-05 08:50:58.713 WARNING (MainThread) [custom_components.mypyllant.datetime] No system data, skipping date time entities
2024-03-05 08:50:58.713 WARNING (MainThread) [custom_components.mypyllant.number] No system data, skipping number entities
2024-03-05 08:50:58.714 WARNING (MainThread) [custom_components.mypyllant.sensor] No system data, skipping sensors
2024-03-05 08:50:58.714 DEBUG (MainThread) [custom_components.mypyllant.sensor] Daily data: None
2024-03-05 08:50:58.714 WARNING (MainThread) [custom_components.mypyllant.sensor] No daily data, skipping sensors
2024-03-05 08:50:58.715 WARNING (MainThread) [custom_components.mypyllant.switch] No system data, skipping switch entities
2024-03-05 08:50:58.715 WARNING (MainThread) [custom_components.mypyllant.water_heater] No system data, skipping water heater

```

**Comment by mzwanzi** [comment]
had the same Problem, also a reload did not help at the first time.
Did the reload again at 09:05 AM and now it works again, maybe a problem with the vaillant api itself

**Comment by pabloromeu** [comment]
Hi, same error here, this line is telling you it's an API problem. Error 500 is always a server error.
```
2024-03-05 08:36:14.894 ERROR (MainThread) [custom_components.mypyllant.coordinator] Error fetching myVAILLANT data: 500, message='Internal Server Error', url=URL('https://api.vaillant-group.com/service-connected-control/end-user-app-api/v1/emf/v2/899af599-23e9-46b6-9988-3cc3e37cb006/currentSystem')
```

**Comment by didierhassio** [comment]
Hello;
Yesterday I had an update on Android to version 2.14.0.
This is surely the cause.

**Comment by Sh0cK9** [comment]
Hello,
i have same issue. I had similar issue in app on my phone (iOS) all informations and settings were unavailable but i have done update of "myVAILLANT" app and application is OK now. But in HASS same issue - "no longer provided by the integration".

**Comment by switschel** [comment]
Seems like vaillant server issues or change of API with the newest App version.

**Comment by fodorg01** [comment]
I can add that today, starting with approx 0:30 am, the server connection seems to be very unstable (seen from HA, I haven't checked the app). There are many server errors (500), and for longer periods the corresponding sensors are unavailable.
But it does recover time-to-time, so the server is sometimes still reachable. So (for now) it doesn't seem to be a breaking API change (then it would be consistently unreachable), but just server accessibility problems.
It was exactly one week ago when there were also server accessibility problems. Maybe Vaillant is doing some maintenance on Tuesdays?

**Comment by aleksander000** [comment]
This integration is still off, but my Android myVAILLANT works correctly

**Comment by mzwanzi** [comment]
works now since about 2 hours, integration reload was required

**Comment by aleksander000** [comment]
yes, thanks

**Comment by ThomasBergholdWieser** [comment]
Its coming up and down again (down again right now) so i think the API is to blame here. I am probably just a bit confused how integrations work when something like this happens. What i mean is why it suddenly reports all entities as "not provided", as they are in fact provided, but just not available right now since the API is down. This is confusing to me (as a user) since this seems a bit misleading. Might be a general home assistant "problem", since the same thing happens with eg. rainbird if it fails to connect to HA for a short time.

However i think this should be closed since its probably not an issue with the integration. But i'll let the repo owner decide what to do here.

**Comment by Brudertac** [comment]
Looks like this is still a Problem. 
I can´t get it back to Work. Since 2-3 Days i have no Data. Reload etc... Nothing works.

**Comment by Brudertac** [comment]
@signalkraft 

Any news on this? Do you need more Data from us?

Thanks!

**Comment by mzwanzi** [comment]
@Brudertac 
i had some problem this week, but since yesterday everything works fine in my scenario...

**Comment by signalkraft** [comment]
@Brudertac are things working again for you?

**Comment by Brudertac** [comment]
No, but i opened a new Case here: https://github.com/signalkraft/mypyllant-component/issues/142

Thanks.
